ios/RCTOneSignal/RCTOneSignal.m

```diff
@@ -145,17 +63,47 @@ - (NSDictionary *)jsonObjectWithString:(NSString *)jsonString {
     NSError *jsonError;
     NSData *data = [jsonString dataUsingEncoding:NSUTF8StringEncoding];
     NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingMutableContainers error:&jsonError];
-    
+
     if (jsonError) {
         [OneSignal onesignal_Log:ONE_S_LL_ERROR message:[NSString stringWithFormat:@"Unable to serialize JSON string into an object: %@", jsonError]];
         return nil;
     }
-    
+
     return json;
 }
 
 - (void)sendEvent:(NSString *)eventName withBody:(NSDictionary *)body {
     [RCTOneSignalEventEmitter sendEventWithName:eventName withBody:body];
 }
 
+- (void)onOSSubscriptionChanged:(OSSubscriptionStateChanges * _Nonnull)stateChanges {
+    [self sendEvent:OSEventString(SubscriptionChanged) withBody:stateChanges.to.toDictionary];
+}
+
+- (void)onOSEmailSubscriptionChanged:(OSEmailSubscriptionStateChanges * _Nonnull)stateChanges {
+    // Example of detecting subscribing to OneSignal
+    if (!stateChanges.from.subscribed && stateChanges.to.subscribed) {
+        //Subscribed for OneSignal push notifications!
+    }
+
+    [self sendEvent:OSEventString(EmailSubscriptionChanged) withBody:stateChanges.to.toDictionary];
+}
+
+- (void)didBeginObserving {
```

Comment:
This method looks like it isn't being call anymore. Do we still need it?

Reply:
I don't thinks so

Reply:
Removed.